caster.go

```diff
@@ -0,0 +1,33 @@
+package qkit
+
+import (
+	"encoding/json"
+)
+
+// Converts any type to a given type. If conversion fails, it returns the zero value of the given type.
+func Cast[T any](val any) T {
+	if val, ok := val.(T); ok {
+		return val
+	}
+	var zero T
+	return zero
+}
+
+// Converts any type to a given type. This partially fills the target in case the 2 types are not directly compatible.
```

Comment:
hmm maybe CastJSON is better name here, since we should know as the user that this works on json representation of the object.

example:
type A struct {
    X string `json:"x"`
    OtherKey string `json:"otherKey"`
}

type B struct {
   X string `json:"xKey"`
   TheX string `json:"x"`
}

if you case A to B, the field `TheX` of B will be filled by `X` of A.